### ISSUE TITLE
bv incorrect answer on LRA formula

### ISSUE THREAD

**Issue by muchang** [Z3Prover/z3#3103]
Hi,
For this formula:
```
(assert (forall ((a Real)) (exists ((b Real)) (and (>= (+ a b) 0.0) (<= b 0.0)))))
(check-sat-using bv)
```
Z3  incorrectly gives sat, while this formula should be unsat.

OS: Ubuntu 18.04
Commit: ff3baff

**Comment by NikolajBjorner** [comment]
This relates to another bug that traces back to the macro finder.
The macro-finder tactic produces a goal that does not preserve unsat.

**Comment by NikolajBjorner** [comment]
fixed with #3705
